### PR TITLE
Sfitz pass idx to bam2pileup

### PR DESCRIPTION
**Pull request by sorelfitzgibbon**
# Description
Three `SomaticSniper` processes were each indexing the reference genome.  Now the index is passed in with the genome.

## Testing Results
`nftest run a_mini-somaticsniper`
output: `/hot/software/pipeline/pipeline-call-sSNV/Nextflow/development/unreleased/sfitz-pass-idx-to-bam2pileup`
log: `/hot/software/pipeline/pipeline-call-sSNV/Nextflow/development/unreleased/sfitz-pass-idx-to-bam2pileup/log-nftest-20231221T210246Z.log`

Comparison of logfiles from a previous run and this run show the index is no longer being re-created:
earlier branch:
`grep -c 'build FASTA index' /hot/software/pipeline/pipeline-call-sSNV/Nextflow/development/unreleased/sfitz-add-F8/a_mini-somaticsniper/call-sSNV-7.0.0/S2-v1.1.5/log-call-sSNV-7.0.0-20231214T233859Z/process-log/SomaticSniper-1.0.5.0/*/log.command.log`
current branch:
`grep -c 'build FASTA index' /hot/software/pipeline/pipeline-call-sSNV/Nextflow/development/unreleased/sfitz-pass-idx-to-bam2pileup/a_mini-somaticsniper/call-sSNV-8.0.0-rc.1/S2-v1.1.5/log-call-sSNV-8.0.0-rc.1-20231221T210251Z/process-log/SomaticSniper-1.0.5.0/*/log.command.log`

# Checklist

- [x] I have read the [code review guidelines](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3187646/Code+Review+Guidelines) and the [code review best practice on GitHub check-list](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3189956/Code+Review+Best+Practice+on+GitHub+-+Check+List).

- [x] I have reviewed the [Nextflow pipeline standards](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3193890/Nextflow+pipeline+standardization).

- [x] The name of the branch is meaningful and well formatted following the [standards](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3189956/Code+Review+Best+Practice+on+GitHub+-+Check+List), using \[AD_username (or 5 letters of AD if AD is too long)]-\[brief_description_of_branch].

- [x] I have set up or verified the branch protection rule following the [github standards](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3190380/GitHub+Standards#GitHubStandards-Branchprotectionrule) before opening this pull request.

- [x] I have added my name to the contributors listings in the ``manifest`` block in the `nextflow.config` as part of this pull request; I am listed already, or do not wish to be listed. (*This acknowledgement is optional.*)

- [x] I have added the changes included in this pull request to the `CHANGELOG.md` under the next release version or unreleased, and updated the date.

- [x] I have updated the version number in the `metadata.yaml` and `manifest` block of the `nextflow.config` file following [semver](https://semver.org/), or the version number has already been updated. (*Leave it unchecked if you are unsure about new version number and discuss it with the infrastructure team in this PR.*)

- [x] I have tested the pipeline on at least one A-mini sample.

### CODE REVIEW

File: config/F2.config [hunk 1]
```diff
@@ -28,10 +28,8 @@ process {
     }
     withName: plot_VennDiagram_R {
         cpus = 2
-        memory = 1.GB
```
Comment (tyamaguchi-ucla) [inline config/F2.config:31]: Is there any reason this line was removed?

File: config/F16.config [hunk 2]
```diff
@@ -8,49 +8,67 @@ process {
...
     withName: call_sIndel_Manta {
-        cpus = 16
-        memory = 15.GB
+        cpus = 6
+        memory = 6.GB
         retry_strategy {
             memory {
                 strategy = 'add'
-                operand = 5.GB
+                operand = 3.GB
```
Comment (tyamaguchi-ucla) [inline config/F16.config:45]: There are many unrelated changes in the `FX.config` in this PR that don't match the PR title or SomaticSniper. Maybe we could either incorporate the changes in https://github.com/uclahs-cds/pipeline-call-sSNV/pull/256 or open a separate PR for them?

Comment (sorelfitzgibbon) [reply config/F16.config]: Sorry @tyamaguchi-ucla, those changes were in the other branch, but I had failed to update it here.

Comment (tyamaguchi-ucla) [reply config/F16.config]: Looks good to me! Maybe merge this branch into the main branch instead of #256?

**Review by yashpatel6 — COMMENTED** [review]
Changes look good! We may want to merge #256 into main and then re-base this PR onto main

File: config/F16.config (hunk 2 above)
Comment (yashpatel6) [reply config/F16.config]: Either order of merge into main should work, as long as we do: merge one branch into main -> update remaining PR and test -> merge into main
